### PR TITLE
Background audio player for acts

### PR DESCRIPTION
**Pull request by GreenRaccoon23**
VideoPlayer class can take an audioId parameter.
If it receives one, it creates a nested audioPlayer property inside iteself.
This audioPlayer property is also a VideoPlayer class.
When VideoPlayer.create() is called, it also calls audioPlayer.create().
The length of the audioPlayer is cut down to match that of the parent VideoPlayer.

The server schema was added to support an audioId, audioStart, audioEnd, and audioVolume properties.

In the html, the audioPlayer is assigned to an element with a display of none, so that it doesn't show on the page.

### CODE REVIEW

File: client/services/videoPlayer.js [hunk 1]
```diff
@@ -15,6 +22,29 @@ angular.module('storyBoard.videoPlayer', ['storyBoard.player'])
     var VIDEO_HEIGHT = 200;
     var VIDEO_WIDTH = 356;
     while( ! window.youtubeApiLoadedAndReady){}
+
+    if (storyFrame.audioId) {
```
Comment (jd009) [inline client/services/videoPlayer.js]: Is this the same piece of information as this.audioId?

Comment (GreenRaccoon23) [inline client/services/videoPlayer.js]: Yes it is. Should I change it to `this.audioId`?

File: client/services/videoPlayer.js [hunk 2]
```diff
@@ -15,6 +22,29 @@ angular.module('storyBoard.videoPlayer', ['storyBoard.player'])
...
+        end: storyFrame.audioEnd,
+        playerDiv: storyFrame.playerDiv + 'Audio',
+        player: {}
+      };
+
+      this.audioPlayer.create(
+        audioStoryFrame, readyCallback,
+        endPlaybackCallback, playingCallback
+      );
+    }
```
Comment (jd009) [inline client/services/videoPlayer.js]: This whole block of code should be wrapped up into a function b/c it is mostly optional code and create() is starting to get pretty long.

File: client/services/videoPlayer.js [hunk 3]
```diff
@@ -15,6 +22,13 @@ angular.module('storyBoard.videoPlayer', ['storyBoard.player'])
     var VIDEO_HEIGHT = 200;
     var VIDEO_WIDTH = 356;
     while( ! window.youtubeApiLoadedAndReady){}
+
+    if (this.audioId) {
+      this.createAudioPlayer(
+        storyFrame, readyCallback, endPlaybackCallback, playingCallback
```
Comment (jd009) [inline client/services/videoPlayer.js]: Will this be a double-call on the next frame due to both having the same endPlaybackCallback?

File: client/services/videoPlayer.js [hunk 4]
```diff
@@ -40,18 +54,48 @@ angular.module('storyBoard.videoPlayer', ['storyBoard.player'])
     this.volume = storyFrame.volume;
   };
 
+  VideoPlayer.prototype.createAudioPlayer = function(storyFrame, readyCallback, endPlaybackCallback, playingCallback){
```
Comment (jd009) [inline client/services/videoPlayer.js]: Could we rename this _createAudioPlayer to signify it is a _private function that should only be called internally by this class?

**Pull request merged**